python/ray/air/execution/resources/virtual.py

```diff
@@ -0,0 +1,31 @@
+from typing import Dict, List, Union
+
+from dataclasses import dataclass
+
+import ray
+from ray.air.execution.resources.fixed import FixedResourceManager
+from ray.air.execution.resources.request import AllocatedResource
+from ray.util.annotations import DeveloperAPI
+
+
+@DeveloperAPI
+@dataclass
+class VirtualAllocatedResource(AllocatedResource):
+    bundles: List[Dict[str, float]]
+
+    def annotate_remote_objects(
+        self, objects
+    ) -> List[Union[ray.ObjectRef, ray.actor.ActorHandle]]:
+        return objects
+
+
+@DeveloperAPI
+class VirtualResourceManager(FixedResourceManager):
```

Comment:
Still a bit abstract for me to understand how this is supposed to used (in testing maybe?) Should we remove it till a later PR?